Remotion/Data/DomainObjects/Persistence/StorageProviderManager.cs

```diff
@@ -45,38 +46,38 @@ public void Dispose ()
   {
     if (!_disposed)
     {
-      if (_storageProviders != null)
-        _storageProviders.Dispose();
+      foreach (var keyValue in _storageProviders!)
+      {
+        keyValue.Value.Dispose();
+      }
 
       _storageProviders = null;
-
       _disposed = true;
-      GC.SuppressFinalize(this);
     }
   }
 
   #endregion
 
-  public StorageProvider GetMandatory (string storageProviderID)
+  public IStorageProvider GetMandatory (string storageProviderID)
   {
     CheckDisposed();
     ArgumentUtility.CheckNotNullOrEmpty("storageProviderID", storageProviderID);
 
-    if (_storageProviders.Contains(storageProviderID))
-      return _storageProviders[storageProviderID]!;
+    if (_storageProviders.TryGetValue(storageProviderID, out var storageProvider))
+      return storageProvider!;
```

Comment:
`storageProvider` cannot be `null`, because we have a `Dictionary<string, IReadOnlyStorageProvider>`, where the value type is non-nullable, so the `!` does not do anything.